src/main/java/io/sqreen/powerwaf/Additive.java

```diff
@@ -83,9 +85,16 @@ public Powerwaf.ResultWithData run(Map<String, Object> parameters,
                     if (!online) {
                         throw new IllegalStateException("This Additive is no longer online");
                     }
-                    ByteBuffer bb;
+                    ByteBuffer persistentBuffer = null;
+                    ByteBuffer ephemeralBuffer = null;
                     try {
-                        bb = this.lease.serializeMore(limits, parameters);
+                        if (persistentData != null) {
+                            persistentBuffer = this.lease.serializeMore(limits, persistentData);
+                        }
+                        if (ephemeralData != null) {
+                            ByteBufferSerializer.ArenaLease ephemeralLease = ByteBufferSerializer.getBlankLease();
```

Comment:
refactored code to ensure lease is freed

Reply:
Seems good now. The ephemeral serialization could be moved before the synchronized block because it doesn't touch any thread-shared resources, but that's a minor thing.